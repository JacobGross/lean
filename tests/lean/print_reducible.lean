prelude

definition id₁ [reducible] {A : Type} (a : A) := a
definition id₂ [reducible] {A : Type} (a : A) := a

definition id₄ [quasireducible] {A : Type} (a : A) := a
definition id₃ [quasireducible] {A : Type} (a : A) := a

definition id₅ [irreducible] {A : Type} (a : A) := a
definition id₆ [irreducible] {A : Type} (a : A) := a

definition pr [reducible] {A B : Type} (a : A) (b : B) := a
definition pr2 {A B : Type} (a : A) (b : B) := a

print [reducible]
print "-----------"
print [quasireducible]
print "-----------"
print [irreducible]
