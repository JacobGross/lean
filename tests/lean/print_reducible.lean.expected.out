id₁
id₂
pr
-----------
id₃
id₄
-----------
id₅
id₆
